src/rooms/UserAdminRoom.ts

```diff
@@ -50,7 +50,7 @@ export class UserAdminRoom {
     }
 
     public async handleLogin() {
-        if (!this.main.oauth2 || this.main.config.puppeting?.enabled) {
+        if (!this.main.oauth2 || !this.main.config.puppeting?.enabled) {
```

Comment:
This condition holds a lot of complexity with the negation of an optional object with a boolean inside.
But according to my testground, it's now doing what you expect it to do.
![Screenshot_2020-05-26_17-22-20](https://user-images.githubusercontent.com/10872136/82918847-92235300-9f75-11ea-8ec2-1a54f33827ac.png)